packages/template-retail-react-app/app/components/link/index.jsx

```diff
@@ -8,20 +8,16 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import {Link as ChakraLink} from '@chakra-ui/react'
 import {Link as SPALink, NavLink as NavSPALink} from 'react-router-dom'
-import {buildPathWithUrlConfig} from '../../utils/url'
 import useSite from '../../hooks/use-site'
 import useLocale from '../../hooks/use-locale'
+import useAppConfig from '../../hooks/use-app-config'
 
 const Link = React.forwardRef(({href, to, useNavLink = false, ...props}, ref) => {
     const _href = to || href
-    const site = useSite()
-    const locale = useLocale()
-
-    // if alias is not defined, use site id
-    const updatedHref = buildPathWithUrlConfig(_href, {
-        locale: locale.alias || locale.id,
-        site: site.alias || site.id
-    })
+    const {site} = useSite()
+    const {locale} = useLocale()
+    const appConfig = useAppConfig()
+    const updatedHref = appConfig.urlTemplateLiteral(_href, site.alias || site.id, locale.id)
```

Comment:
Thing to consider: Maybe we can memoize this func? It might not be worth it since it is cheap to construct a string template. 🤔 
```suggestion
     const updatedHref = React.useMemo(
        () => appConfig.urlTemplateLiteral(_href, site.alias || site.id, locale.id),
        [site.alias, site.id, locale.id, _href]
    )```

Reply:
Typically `React.useMemo` is used to address computationally expensive operations. 
Memoization “remembers” the result of a complex function when the same parameters are used. 

In our case, we provide three string values to a method with a few ternaries inside a template string. This is not an expensive operation, thus we don't get a benefit from using `useMemo`.

I measured the CPU Profiler using `useMemo`. I don't see any improvement and it looks like it even harms slightly the performance. I'm guessing it's because is adding complexity to an already simple operation.